modules/ROOT/pages/client-id-based-policies.adoc

```diff
@@ -14,48 +14,73 @@ endif::[]
 * xref:policy-ping-federate.adoc[PingFederate OAuth 2.0 Token Enforcement]
 * xref:policy-openid-connect.adoc[OpenID Connect OAuth 2.0 Token Enforcement]
 
-= Offline mode
+== Policy Prerequisites and Restrictions
 
-Credentials data from client application credentials registered to protected APIs, are provided by Anypoint Platform. Once a policy requiring such information is applied (eg Client ID enforcement policy), the Runtime caches persistently the information locally updating it periodically.
+For the policy to work successfully, you must:
 
-Besides its performance benefits, this action enables the policy to work offline in the event of a sudden loss of connectivity to the platform. The enforcement is applied regardless of the state of the management plane, with existing data. Once connectivity is re-established, the cache will be periodically updated again.
+* Generate the client ID and client secret credentials from Anypoint Platform.
++
+Using client credentials generated outside of Anypoint Platform fails the HTTP request.
+* Provide the client ID and client secret pair of the registered client application.
+* If using REST console, create a trait in the RAML root and then reference it in every operation of your API affected by policy.
++
+To obtain the RAML code to add to the policy, go to the corresponding policy in your API Manager.
+//It looks like we might need this for other policies too? If so, we should move it out of here and put it in a central place.
++
+image::raml-snippet.png[]
 
-= REST Console
-To prevent user requests from being rejected when using the REST Console, create a trait in the RAML root and then reference it in every operation of your API affected by policy. You may find the RAML snippet link containing the RAML code you need to add to the RAML in the corresponding policy from the list of applied policies in the Policies tab of your API specification in API Manager.
 
-image::raml-snippet.png[]
+== Policy Implementation Best Practices
+
+If you need to track client application access to your API, apply the Client ID Enforcement policy to makes sure that the client credentials are provided in all HTTP requests. 
+
+//This may be too little info to warrant a section of its own. Let's reassess.
 
-= Credentials source
 
-The policy can be configured to extract the client ID and client secret from the HTTP request in different ways using DataWeave 2.0 expressions.
+== Configure the Policy
 
-The following examples assume that the values for application credentials are '1234' for client ID and 'abcd' for client secret.
+You can configure the policy to extract the client ID and client secret from the HTTP request in different ways using DataWeave 2.0 expressions. If when configuring the policy, the client secret DataWeave 2.0 expression is not specified, then only the client ID specified in the HTTP request is validated against the protected resource.
```

Comment:
I would rephrase this one, the section will be Configure the policy and then the different ways to do it right? Then I suggest:

The Client ID enforcement policy can be configured to obtain the client credentials in different ways. One option is to configure the policy to expect the credentials as part of the Authorization header by selecting the HTTP Basic Authentication option, in this case the application consuming the API needs to use basic authentication to send the credentials in the requests. 

The other alternative is to select Custom Expression, this is the default selection and it allows the customer to provide an expression to decide where to extract the credentials from the request. In Mule 3 a customer can provide a MEL Expression, in Mule 4 it should be a DataWeave 2.0 expression [not sure if this page is just for 4, 3 or shared). This option is needed when the customer wants to send the credentials in a custom header or somewhere else in the payload. 

There is also the option to set the client secret as optional by leaving the field blank. 

Following are some configuration examples:

[I would leave the invalid scenario after the configuration examples]

Reply:
Done! Thanks!